### PR TITLE
Removing defaults from CF specific property keys

### PR DESCRIPTION
**Pull request by sobychacko**
Renaming memory, disk and services property keys so that the word defaults is removed from them.

### CODE REVIEW

**Comment by ericbottard** [comment]
If we remove the `.defaults` part, then this will clash with what is used in `CloudFoundryDeployerProperties`.
Maybe the naming is wrong, and it's what inside `CloudFoundryDeployerProperties` that should be moved inside a `Defaults` inner field, but I think this PR as is should not be merged.

**Comment by sobychacko** [comment]
@ericbottard The origin of this PR is this issue: https://github.com/spring-cloud/spring-cloud-dataflow-server-cloudfoundry/issues/99. 
Not sure entirely how this clash would occur though as `CloudFoundryAppDeployer` would query for the property from env variables which by that time any configuration properties provided through `CloudFoundryDeployerProperties` are all bound. No?

**Comment by gregturn** [comment]
It would break spring-cloud-spinnaker due to the UI supplying such a named property and can't fetch it from the code.

**Comment by markpollack** [comment]
Note that the other deployers are not using the defaults field, so one way or another we need some changes.  I'll take a closer look later.

**Comment by markpollack** [comment]
If we need to have another level (and insert one for the other deployers) the word 'app' or 'application' would read more naturally to me...

**Comment by ericbottard** [comment]
Actually, maybe the best thing to do is to split `CloudFoundryDeployerProperties` in two: stuff related to connection, etc... and stuff related to defaults for a deployment.

**Comment by markpollack** [comment]
There is no clash functionally speaking (no side effects) but something is sort of mismatched here.

When the server starts, it will use  `spring.cloud.deployer.cloudfoundry.memory`  defined in [CloudFoundryDeployerProperties](https://github.com/spring-cloud/spring-cloud-deployer-cloudfoundry/blob/master/src/main/java/org/springframework/cloud/deployer/spi/cloudfoundry/CloudFoundryDeployerProperties.java)  as default value for memory when deploying an app.  

To override this when deploying an application, some key needs to be passed in as a deployment property.  These deployment properites are currently contained in `AppDeploymentRequest.getEnvironmentProperties`.  The name seems confusing to me as they aren't environment variables that get passed on to the app (javadocs are clear) but we aren't going to change the name now I'd bet. 

Anywa, it does so happen that we are looking for a property of the same name in the AppDeploymentRequest, to override this default.  However, we do treat the instance count as special in the deployer, for example

```dataflow:> stream deploy --name ticktock --properties "app.time.count=3"```

And this feels like the right way, in particular to qualify the memory setting to a specific app in the stream.  The current approach will set up the memory for all the apps, not each individual one.  

Memory and Disk size might be common enough that we want to get them added into `AppDeployer` interface (we have group and count there) which ATM have the keys, `spring.cloud.deployer.count` but that gets translated from `app.time.count` at the DSL level.  

In short, having `app.time.memory` and `app.time.disk` as the DSL level would be ideal. 

However, if we don't make that change, going with `spring.cloud.deployer.cloudfoundry.memory` isn't so bad.  An alternative would be to add `apps` for the memory and disk key name, e.g. `spring.cloud.deployer.cloudfoundry.apps.memory`

**Comment by markpollack** [comment]
I'm going to merge this as it is, updating the readme to make mention of these `@ConfigurationProperties`  and create a new issue in spring-cloud-deployer.

**Comment by markpollack** [comment]
Merged as 37cbaea76760189314bb58091b6f2489f7812b12
